modules/pipeline/providers/trigger/trigger.service.go

```diff
@@ -0,0 +1,224 @@
+	pipelineSvc        *pipelinesvc.PipelineSvc
+}
+
+func (s *TriggerService) SetPipelineSvc(pipelineSvc *pipelinesvc.PipelineSvc) {
+	s.pipelineSvc = pipelineSvc
+}
+
+func (s *TriggerService) RunPipelineByTriggerRequest(ctx context.Context, req *pb.PipelineTriggerRequest) (*pb.PipelineTriggerResponse, error) {
+	err := s.checkPermission(ctx)
+	if err != nil {
+		return nil, apierrors.ErrCheckPermission.AccessDenied()
+	}
+
+	pipelineTriggers, err := s.triggerDbClient.ListPipelineTriggers(req)
+	if err != nil {
+		return nil, err
+	}
+
+	// Get trigger label
+	triggerLabel := make(map[string]string)
+	for key, val := range req.Label {
+		triggerLabel[fmt.Sprintf("triggers.%s.%s", req.EventName, key)] = val
+	}
+
+	// De-duplication
+	pipelineTriggersMap := make(map[uint64]triggerDb.PipelineTrigger)
+	for _, trigger := range pipelineTriggers {
+		if _, ok := pipelineTriggersMap[trigger.PipelineDefinitionID]; !ok {
+			pipelineTriggersMap[trigger.PipelineDefinitionID] = trigger
+		}
+	}
+
+	var pipelineIDs []uint64
+	for _, trigger := range pipelineTriggersMap {
+
+		pipelineDefinition, err := s.definitionDbClient.GetPipelineDefinitionByNameAndSource(trigger.PipelineSource, trigger.PipelineYmlName)
+		if err != nil {
+			return nil, err
+		}
+
+		if pipelineDefinition.PipelineYmlName != "" && pipelineDefinition.PipelineYml != "" && pipelineDefinition.PipelineSource != "" {
+			if pipelineDefinition.Extra.CreateRequest == nil {
+				break
+			}
+
+			label := make(map[string]string)
+			if pipelineDefinition.Extra.CreateRequest != nil && pipelineDefinition.Extra.CreateRequest.Labels != nil {
+				label = pipelineDefinition.Extra.CreateRequest.Labels
+			}
+			for key, val := range triggerLabel {
+				label[key] = val
+			}
+
+			pipeline, err := s.pipelineSvc.CreateV2(&apistructs.PipelineCreateRequestV2{
+				PipelineYml:            pipelineDefinition.PipelineYml,
+				ClusterName:            pipelineDefinition.Extra.CreateRequest.ClusterName,
+				PipelineYmlName:        pipelineDefinition.PipelineYmlName,
+				PipelineSource:         pipelineDefinition.PipelineSource,
+				Labels:                 label,
+				NormalLabels:           pipelineDefinition.Extra.CreateRequest.NormalLabels,
+				ConfigManageNamespaces: pipelineDefinition.Extra.CreateRequest.ConfigManageNamespaces,
+				AutoRunAtOnce:          true,
+				AutoStartCron:          false,
+				ForceRun:               true,
+				IdentityInfo: apistructs.IdentityInfo{
+					UserID: apis.GetUserID(ctx),
+				},
+			})
+			if err != nil {
+				return nil, err
+			}
+			pipelineIDs = append(pipelineIDs, pipeline.ID)
+		}
+	}
+
+	return &pb.PipelineTriggerResponse{PipelineIDs: pipelineIDs}, nil
+}
+
+func (s *TriggerService) RegisterTriggerEvent(definition definition.PipelineDefinitionProcess, yml pipelineyml.PipelineYml) error {
+
+	newPipelineTriggerMap := GetPipelineTriggerMap(yml)
+
+	oldPipelineTriggers, err := s.triggerDbClient.GetPipelineTriggerByID(definition.ID)
```

Comment:
The method parameter definition itself is the latest
same like db value

Reply:
done